Assignment_05.rb

```diff
@@ -120,19 +198,32 @@ def process_input
       total_time_in_min: [],
       occupation_percecntage: []
     }
-
-    tasks.each_with_index do |task, index|
-      report[:categories] << task.name
-      report[:total_time_in_min] << task.total_time_in_minute
-      report[:occupation_percecntage] << task.occupation_percecntage_day_wise(report[:total_time_in_min][index])
-    end
     
+    begin
+      tasks.each_with_index do |task, index|
+        report[:categories] << task.name
+        report[:total_time_in_min] << task.total_time_in_minute
+        report[:occupation_percecntage] << task.occupation_percecntage_day_wise(report[:total_time_in_min][index])
+      end
+    rescue => e
+      puts "Error occured in process_input: #{e.message}"
+    end
+
     report
   end
+
+  private
+
+  def validate_inputs(task_attributes)
+    raise "Attribute id of Task is not present" if task_attributes[:id].nil?
+    raise "Attribute name of Task is not present" if task_attributes[:name].nil?
+    raise "Attribute time_logs of Task is not present" if task_attributes[:time_logs].nil?
+    raise "Task with id: #{task_attributes[:id]} alredy present" if self.tasks.any? { |task| task.id == task_attributes[:id] }
```

Comment:
this part also should go under id validation for checking uniqueness

Reply:
seems you have already those validation present inside validates_id, validates_name, validates_time_logs methods? can we reuse them?